packages/devtools_app/lib/src/analytics/_analytics_web.dart

```diff
@@ -275,6 +275,60 @@ void screen(
   );
 }
 
+String _timedOperationKeyHelper(String screenName, String timedOperation) {
+  return '$screenName-$timedOperation';
+}
+
+final _timedOperationsInProgress = <String, DateTime>{};
+
+// Use this method coupled with `timeEnd` when an operation cannot be timed in
+// a callback, but rather needs to be timed instead at two disjoint start and
+// end marks.
+void timeStart(String screenName, String timedOperation) {
+  final startTime = DateTime.now();
+  final operationKey = _timedOperationKeyHelper(
+    screenName,
+    timedOperation,
+  );
+  _timedOperationsInProgress[operationKey] = startTime;
+}
+
+// Use this method coupled with `timeStart` when an operation cannot be timed in
+// a callback, but rather needs to be timed instead at two disjoint start and
+// end marks.
+void timeEnd(
+  String screenName,
+  String timedOperation, {
+  ScreenAnalyticsMetrics Function() screenMetricsProvider,
+}) {
+  final endTime = DateTime.now();
+  final operationKey = _timedOperationKeyHelper(
+    screenName,
+    timedOperation,
+  );
+  final startTime = _timedOperationsInProgress[operationKey];
```

Comment:
remove returns the value so you can combine this with the .remove at the end of the file.

Reply:
done